contracts/stark-proxy/v3/impl/SP3Owner.sol

```diff
@@ -0,0 +1,335 @@
+    );
+    uint256 elapsed = block.timestamp.sub(timestamp);
+    require(
+      elapsed >= FORCED_TRADE_WAITING_PERIOD,
+      'SP2Owner: Waiting period has not elapsed for forced trade'
+    );
+    require(
+      elapsed <= FORCED_TRADE_WAITING_PERIOD.add(FORCED_TRADE_GRACE_PERIOD),
+      'SP2Owner: Grace period has elapsed for forced trade'
+    );
+    _QUEUED_FORCED_TRADE_TIMESTAMPS_[argsHash] = 0;
+    _forcedTradeRequest(args, signature, false);
+  }
+
+  /**
+   * @notice Request to cancel a pending deposit to the exchange.
+   *
+   * @param  starkKey   The STARK key of the account. Must be authorized by OWNER_ROLE.
+   * @param  assetType  The exchange asset ID for the deposit.
+   * @param  vaultId    The exchange position ID for the deposit.
+   */
+  function depositCancel(
+    uint256 starkKey,
+    uint256 assetType,
+    uint256 vaultId
+  )
+    external
+    nonReentrant
+    onlyRole(OWNER_ROLE)
+    onlyAllowedKey(starkKey)
+  {
+    _depositCancel(starkKey, assetType, vaultId, false);
+  }
+
+  /**
+   * @notice Reclaim a canceled pending deposit to the exchange. Requires that `depositCancel`
+   *  was previously called.
+   *
+   * @param  starkKey   The STARK key of the account. Must be authorized by OWNER_ROLE.
+   * @param  assetType  The exchange asset ID for the deposit.
+   * @param  vaultId    The exchange position ID for the deposit.
+   */
+  function depositReclaim(
+    uint256 starkKey,
+    uint256 assetType,
+    uint256 vaultId
+  )
+    external
+    nonReentrant
+    onlyRole(OWNER_ROLE)
+    onlyAllowedKey(starkKey)
+  {
+    _depositReclaim(starkKey, assetType, vaultId, false);
+  }
+
+  /**
+   * @notice Delegates governance power of DYDX or stakedDYDX to a delegatee.
+   *
+   * @param  token     The address of the token to delegate. Must be DYDX_ADDRESS or STAKED_DYDX_ADDRESS.
+   * @param  delegatee The address to delegate to.
+   */
+  function delegate(
+    address token,
+    address delegatee
+  )
+    external
+    nonReentrant
+    onlyRole(OWNER_ROLE)
+  {
+    require(
+      token == DYDX_ADDRESS ||
+      token == STAKED_DYDX_ADDRESS,
+      'tokenAddress is not DYDX or stakedDYDX'
+    );
+    IGovernancePowerDelegationERC20(token).delegate(delegatee);
+  }
+
+  /**
+   * @notice Delegates a specific governance power of DYDX or stakedDYDX to a delegatee.
+   *
+   * @param  token           The address of the token to delegate. Must be DYDX_ADDRESS or STAKED_DYDX_ADDRESS.
+   * @param  delegatee       The address to delegate to.
+   * @param  delegationType  The type of delegation (VOTING_POWER, PROPOSITION_POWER).
+   */
+  function delegateByType(
+    address token,
+    address delegatee,
+    IGovernancePowerDelegationERC20.DelegationType delegationType
+  )
+    external
+    nonReentrant
+    onlyRole(OWNER_ROLE)
+  {
+    require(
+      token == DYDX_ADDRESS ||
+      token == STAKED_DYDX_ADDRESS,
+      'tokenAddress is not DYDX or stakedDYDX'
+    );
```

Comment:
If you want, you can make this a [modifier](https://solidity-by-example.org/function-modifier/), although fine with this method as well.